compute: fix FirewallPolicy observe by correcting identifier resolution

AndresAbdo opened the pull request:
### Description of your changes

This PR fixes observe failures for `ComputeFirewallPolicy` resources caused by
incorrect identifier resolution during observe.

When the identifier was not constructed correctly, the provider issued a list
request instead of a direct get, which resulted in a 400 error due to a missing
`parentId`. This change ensures the correct identifier is used so observe calls
target the proper get endpoint.

This PR also includes regenerated APIs and CRDs produced by `make reviewable`.
As part of regeneration, the `FirewallPolicy` `name` observation field was
removed, since it is no longer exposed by the Terraform provider schema and is
represented via external-name semantics.

Fixes #820

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

- Tested locally using `make local-deploy`
- Verified that an existing FirewallPolicy can be successfully observed
- Confirmed that observe behavior works correctly in cases where the current
  production code fails

AndresAbdo committed:
config: normalize firewall policy name from external-name

Derive google_compute_firewall_policy name from the last segment of
crossplane.io/external-name and omit name from spec to avoid conflicts.
Fixes Observe failures reported in #820.

Signed-off-by: Andres Abdo <andresabdo23@gmail.com>

AndresAbdo committed:
chore: regenerate compute APIs with make reviewable

Regenerate provider APIs and CRDs after FirewallPolicy identifier changes.

This commit includes:
- Removal of the deprecated/omitted FirewallPolicy `name` observation field,
  as it is no longer present in the Terraform provider schema or is represented
  via external-name semantics.
- Updated deepcopy and CRD schemas produced by make reviewable.

No functional changes beyond generated output.

Signed-off-by: Andres Abdo <andresabdo23@gmail.com>

jonasz-lasut reviewed (changes requested):
Hi @AndresAbdo, thank you for the PR and for patiently keeping it alive — and your diagnosis is correct. I verified it at source:

**Confirmed (code, terraform-provider-google `e40b75688dfe` as pinned by `main`):** `google_compute_firewall_policy` builds its Read URL from the computed `name` attribute, not from the TF ID — [`resource_compute_firewall_policy.go#L306`](https://github.com/hashicorp/terraform-provider-google/blob/e40b75688dfe/google/services/compute/resource_compute_firewall_policy.go#L306):

```go
url, err := tpgresource.ReplaceVars(d, config, transport_tpg.BaseUrl(Product, config)+"locations/global/firewallPolicies/{{name}}")
```

`name` is `Computed: true` only (L156), and upjet never runs the TF importer that would parse it out of the ID (L495). So on observe with a pre-set external-name (or any state reconstruction with empty `atProvider`), `{{name}}` is empty, the GET lands on the collection URL — which is the **list** endpoint whose required query parameter is `parentId` (compare the create URL at L227: `firewallPolicies?parentId={{parent}}`) — and GCP returns the reported `Error 400: Required field 'parentId' not specified`. Fresh creates never hit this, which is why the happy path hides it. `SetIdentifierArgumentFn` seeding `name` from the external-name is exactly the right primitive; we use the same pattern for `google_cloud_identity_group` and the apigee resources.

That said, I'd like to request changes on the mechanics — the current shape misses the reported resource and changes the API surface more than the fix needs:

**1. The namespaced scope is untouched, and that is the scope #820 is filed against.** The issue reproduces on the **namespaced** API (`compute.gcp.m.upbound.io/v1beta1`); this PR only adds the configurator to `config/cluster/compute/config.go` and regenerates `apis/cluster/...` + the cluster CRD, so the reporter's MR would still fail after merge.

**2. Please don't remove `status.atProvider.name`.** The `OmittedFields` addition is what drops it from the Observation and the CRD — it isn't required for the fix, and it's a user-visible regression: that field is where consumers read the GCP-allocated numeric policy ID. The PR description's rationale ("no longer exposed by the Terraform provider schema") doesn't hold — the schema still defines `name` ([L156-L160](https://github.com/hashicorp/terraform-provider-google/blob/e40b75688dfe/google/services/compute/resource_compute_firewall_policy.go#L156-L160)); the field disappeared from the generated types because of the `OmittedFields` change, not upstream.

**3. Both of the above resolve themselves if the fix moves to `config/externalname.go`.** That file is shared by the cluster and namespaced providers, external-name helpers are the established home for this exact pattern (`cloudIdentity()`, `apigeeOrganization()`, …), and with no `OmittedFields` there are **zero** generated-API/CRD changes — the whole PR reduces to something like:

```go
// firewallPolicy configures the external name for
// google_compute_firewall_policy. The TF ID is
// locations/global/firewallPolicies/{name}, where `name` is the
// GCP-allocated numeric id exposed only as a computed attribute. The
// resource's Read/Update/Delete build their URLs from `name` rather than
// from the TF ID, so on import / state reconstruction `name` must be
// seeded from the external-name; otherwise Read GETs the list endpoint
// and fails with 400 "Required field 'parentId' not specified" (#820).
func firewallPolicy() config.ExternalName {
	e := config.IdentifierFromProvider
	e.SetIdentifierArgumentFn = func(base map[string]any, externalName string) {
		if externalName == "" {
			return
		}
		parts := strings.Split(externalName, "/")
		base["name"] = parts[len(parts)-1]
	}
	return e
}
```

with the map entry at [`config/externalname.go#L273`](https://github.com/crossplane-contrib/provider-upjet-gcp/blob/733d1461f/config/externalname.go#L273) changed to `"google_compute_firewall_policy": firewallPolicy(),`.

**4. Two nits in the current `SetIdentifierArgumentFn`, for whichever shape you land:** please add the `if externalName == "" { return }` guard the existing helpers all carry (otherwise fresh creates seed `name: ""` into the parameters), and note `len(parts) > 0` is always true — `strings.Split` never returns an empty slice — so the guard as written is dead code.

On testing: your local `make local-deploy` verification of observing an existing policy covers the failing path well; this resource needs an organization parent, so it isn't uptest/e2e-testable in the standard pipeline — local verification is what we can reasonably ask for here. If you can re-run the same check once on the `externalname.go` variant (ideally against the namespaced `FirewallPolicy` kind, since that's the reported GVK), I think this is good to go.

Thanks again — the analysis in this PR is what made the root cause easy to confirm.

AndresAbdo committed:
config: relocate FirewallPolicy external-name fix to shared config

Address review feedback on #875 by moving the google_compute_firewall_policy
identifier fix from config/cluster/compute/config.go into the shared
config/externalname.go, and drop the OmittedFields change.

The resource's Read/Update/Delete build their GCP URLs from the computed
name attribute rather than from the Terraform ID, so on observe with a
pre-set external-name name is empty, the GET hits the firewallPolicies
list endpoint, and GCP returns 400 "Required field 'parentId' not
specified" (#820). A SetIdentifierArgumentFn that seeds name from the
last segment of the external-name resolves this.

Changes vs. the previous approach:
- Move the fix to config/externalname.go via a new firewallPolicy() helper.
  That file is shared by the cluster and namespaced providers, so the fix
  now also covers the namespaced FirewallPolicy (compute.gcp.m.upbound.io),
  which is the GVK #820 was reported against.
- Drop the OmittedFields addition, restoring status.atProvider.name to the
  Observation and CRD. The field is where consumers read the GCP-allocated
  numeric policy id; it is still defined by the upstream schema and was only
  removed as a side effect of OmittedFields, not by the fix itself. This
  reduces the change to zero generated-API/CRD diffs.
- Guard against an empty external-name so fresh creates don't seed name="",
  and drop the dead len(parts) check (strings.Split never returns an empty
  slice).

Verified locally with make local-deploy.config,compute against a
namespaced FirewallPolicy: observe of an existing policy now succeeds
instead of failing with the parentId error.

Signed-off-by: Andres Abdo <andresabdo23@gmail.com>

AndresAbdo committed:
Merge remote-tracking branch 'upstream/main' into bugFix/firewallpolicy-observe-820

Signed-off-by: Andres Abdo <andresabdo23@gmail.com>

jonasz-lasut reviewed (approved):
Thank you @AndresAbdo lgtm!
